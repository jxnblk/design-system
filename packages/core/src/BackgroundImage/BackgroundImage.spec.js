import React from 'react'
import { BackgroundImage } from '..'

describe('BackgroundImage', () => {
  test('renders', () => {
    const json = rendererCreateWithTheme(<BackgroundImage />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with image', () => {
    const json = rendererCreateWithTheme(<BackgroundImage image='hello.png' />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with height', () => {
    const json = rendererCreateWithTheme(<BackgroundImage height='320px' />).toJSON()
    expect(json).toMatchSnapshot()
  })

  describe('variations', () => {
    describe('parallax variation', () => {
      test('default render', () => {
        const json = rendererCreateWithTheme(
          <BackgroundImage variation='parallax' image='hello.png' />
        ).toJSON()
        expect(json).toMatchSnapshot()
        expect(json).toHaveStyleRule('background-attachment', 'fixed')
      })
    })
  })
})
