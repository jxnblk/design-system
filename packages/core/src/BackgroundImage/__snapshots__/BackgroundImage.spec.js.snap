// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImage renders 1`] = `
.c0 {
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-color: #eceff2;
  background-attachment: scroll;
}

<div
  className="c0"
/>
`;

exports[`BackgroundImage renders with height 1`] = `
.c0 {
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-color: #eceff2;
  background-attachment: scroll;
  height: 320px;
}

<div
  className="c0"
  height="320px"
/>
`;

exports[`BackgroundImage renders with image 1`] = `
.c0 {
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-color: #eceff2;
  background-attachment: scroll;
  background-image: url(hello.png);
}

<div
  className="c0"
/>
`;

exports[`BackgroundImage variations parallax variation default render 1`] = `
.c0 {
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-color: #eceff2;
  background-attachment: fixed;
  background-image: url(hello.png);
}

<div
  className="c0"
/>
`;
